packages/cadl-python/src/emitter.ts

```diff
@@ -140,6 +144,18 @@ function handleDiscriminator(program: Program, type: ModelType, model: Record<st
     }
 }
 
+function getEffectiveSchemaType(program: Program, type: ModelType): ModelType {
```

Comment:
we want to use the `getEffectiveModelType` function that @markcowl linked in the discussion, see its use [here](https://github.com/microsoft/cadl/blob/main/packages/openapi3/src/openapi.ts#L10) in the openapi emitter